cancermuts_streamlit/streamlit_utils.py

```diff
@@ -0,0 +1,80 @@
+import streamlit as st
+import os
+import pandas as pd
+
+def filter_dataframe(df: pd.DataFrame) -> pd.DataFrame:
+    addFilter = st.checkbox("Add filter")
+
+    if not addFilter:
+        return df
+    
+    df = df.copy()
+
+    for col in df.columns:
+        try:
+            df[col] = pd.to_datetime(df[col], format='%d-%m-%Y')
+        except Exception:
+            pass

```

Comment:
I'm not sure what this is meant to achieve

Reply:
In order to filter on the date I have to covert to a datetime object, so I test if any of the columns can be converted (basically test if there is a date column). I guess I thought it would be more robust as a function in case it would be used for something else. But I can delete the try statement if it is redundant